### DISCUSSION TITLE
gulpfile.babel.js does not work in subdirectories

### DISCUSSION THREAD

**Discussion by Nantris** [gulpjs/gulp#2620] [Help]
Before you open this issue, please complete the following tasks:

* [x] use the search bar at the top of the page to search this repository for similar issues or discussions that have already been opened.
* [x] if you are looking for help from the gulp team or community, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a problem with the plugin you're using, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a bug in our code, open this issue.

### What were you expecting to happen?

`gulpfile.babel.js` in a subdirectory should be able to use ES6 imports.

### What actually happened?

"SyntaxError: Cannot use import statement outside a module"

### Please give us a sample of your gulpfile

```js
import { getAssetsSubdir } from '@internal/build-utils';

const gulp = require('gulp');
```

### Terminal output / screenshots

```sh
$ pwd
/cygdrive/c/Projects/ourProject/packages/shared/viewer-component
$ gulp copyAssets
C:\Projects\ourProject\packages\shared\viewer-component\gulpfile.babel.js:2
import { getRnAssetsSubdir } from '@internal/build-utils'; //
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:988:16)
    at Module._compile (internal/modules/cjs/loader.js:1036:27)
    at Module._compile (C:\Projects\ourProject\node_modules\pirates\lib\index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Object.newLoader [as .js] (C:\Projects\ourProject\node_modules\pirates\lib\index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at requireOrImport (C:\Users\Remie\AppData\Roaming\npm\node_modules\gulp\node_modules\gulp-cli\lib\shared\require-or-import.js:19:11)

```

### Please provide the following information:
* OS & version [e.g. MacOS Catalina 10.15.4]: Windows 7
* node version (run `node -v`): 12.20.1
* npm version (run `npm -v`): 6.12.x
* gulp version (run `gulp -v`): 4.0.2
* gulp-cli version: 2.3.0

### Additional information
`gulpfile.babel.js` works great at the root of the project, but fails in subdirectories in our monorepo. Any way to make this work? Thank you!

**Comment by phated** [comment]
Moving to GitHub Discussions. Issue tracker is not used for questions.

**Reply by Nantris** [reply]
I feel this constitutes a bug. I filled out the issue template as you requested be done in #2446.

**Reply by Nantris** [reply]
> **if you think there is a bug in our code, open this issue.**

**Reply by phated** [reply]
> Any way to make this work?

1. That's a support question
2. You have no reproduction, so this is being handled as a support question

Hope someone from the community can help you out because I'm unsubscribing.

**Reply by Nantris** [reply]
Frustrating since I followed your explicit request, but okay.

Lesson learned, don't ask for workarounds - just report the bug.
